modules/widgets/milestone/milestone.php

```diff
@@ -0,0 +1,366 @@
+<?php
+/*
+Plugin Name: Milestone
+Description: Countdown to a specific date.
+Version: 1.0
+Author: Automattic Inc.
+Author URI: http://automattic.com/
+License: GPLv2 or later
+*/
+
+class Jetpack_Milestone {
+	public static function init() {
+		add_action( 'widgets_init', array( __class__, 'register_widget' ) );
+	}
+	public static function register_widget() {
+		register_widget( 'Milestone_Widget' );
+	}
+}
+
+Jetpack_Milestone::init();
+
+class Milestone_Widget extends WP_Widget {
+	private static $dir       = null;
+	private static $url       = null;
+	private static $labels    = null;
+	private static $defaults  = null;
+	private static $config_js = null;
+
+	function __construct() {
+		$widget = array(
+			'classname'   => 'milestone-widget',
+			'description' => __( 'Display a countdown to a certain date.', 'jetpack' )
+		);
+
+		$control = array(
+			'width' => 251, // Chrome needs a little extra room for the date fields.
+		);
+
+		parent::__construct(
+			'Milestone_Widget',
+			/** This filter is documented in modules/widgets/facebook-likebox.php */
+			apply_filters( 'jetpack_widget_name', __( 'Milestone', 'jetpack' ) ),
+			$widget,
+			$control
+		);
+
+		self::$dir = trailingslashit( __DIR__ );
+		self::$url = plugin_dir_url( __FILE__ );
+		self::$labels = array(
+			'year'    => __( 'year', 'jetpack' ),
+			'years'   => __( 'years', 'jetpack' ),
+			'month'   => __( 'month', 'jetpack' ),
+			'months'  => __( 'months', 'jetpack' ),
+			'day'     => __( 'day', 'jetpack' ),
+			'days'    => __( 'days', 'jetpack' ),
+			'hour'    => __( 'hour', 'jetpack' ),
+			'hours'   => __( 'hours', 'jetpack' ),
+			'minute'  => __( 'minute', 'jetpack' ),
+			'minutes' => __( 'minutes', 'jetpack' ),
+			'second'  => __( 'second', 'jetpack' ),
+			'seconds' => __( 'seconds', 'jetpack' ),
+		);
+
+		add_action( 'wp_enqueue_scripts', array( __class__, 'enqueue_template' ) );
+		add_action( 'admin_enqueue_scripts', array( __class__, 'enqueue_admin' ) );
+		add_action( 'wp_footer', array( __class__, 'localize_script' ) );
+
+		if ( is_active_widget( false, false, $this->id_base, true ) || is_active_widget( false, false, 'monster', true ) )
+			add_action( 'wp_head', array( __class__, 'styles_template' ) );
+	}
+
+	public static function enqueue_admin( $hook_suffix ) {
+		if ( 'widgets.php' == $hook_suffix )
+			wp_enqueue_style( 'milestone-admin', self::$url . 'style-admin.css', array(), '20111212' );
+	}
+
+	public static function enqueue_template() {
+		wp_enqueue_script( 'milestone', self::$url . 'milestone.js', array( 'jquery' ), '20160520', true );
+	}
+
+	public static function styles_template() {
+		global $themecolors;
```

Comment:
Since that's specific to WordPress.com, should we add a conditional here, and allow site owners to use their own colors? I think it'd be nice, especially since it's not super easy to dequeue that CSS.